src/identity/main.cairo

```diff
@@ -88,17 +174,164 @@ mod Identity {
-            ref self: ContractState, id: u128, field: felt252, data: felt252, domain: felt252
+            ref self: ContractState, id: u128, field: felt252, data: felt252, domain: u32
         ) {
             // todo: when volition comes, handle the domain
-            self.verifier_data.write((id, field, get_caller_address()), data);
+            let verifier = get_caller_address();
+            self.verifier_data.write((id, field, verifier), data);
+            self
+                .emit(
+                    Event::VerifierDataUpdate(
+                        VerifierDataUpdate { id, field, _data: data, verifier, }
+                    )
+                )
+        }
+
+        fn set_extended_verifier_data(
+            ref self: ContractState, id: u128, field: felt252, data: Span<felt252>, domain: u32
+        ) {
+            let verifier = get_caller_address();
+            self
+                .set(
+                    VERIFIER_DATA_ADDR,
+                    array![id.into(), field, verifier.into()].span(),
+                    0,
+                    data,
+                    domain
+                );
+            self
+                .emit(
+                    Event::ExtendedVerifierDataUpdate(
+                        ExtendedVerifierDataUpdate { id, field, _data: data, verifier, }
+                    )
+                );
+        }
+    }
+
+    //
+    // Internals
+    //
+
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
+        // todo: move these functions into a storage contract when components are available
+        fn get_extended(
+            self: @ContractState,
+            fn_name: felt252,
+            params: Span<felt252>,
+            offset: u8,
+            length: felt252,
+            domain: u32
+        ) -> Span<felt252> {
+            let base = self.compute_base_address(fn_name, params);
+            let mut data = ArrayTrait::new();
+            let mut offset = 0;
+            loop {
+                if length == offset.into() {
+                    break ();
+                }
+                let value = self._get(domain, base, offset);
+                data.append(value);
+                offset += 1;
+            };
+            data.span()
+        }
+
+        fn get_unbounded(
+            self: @ContractState, fn_name: felt252, params: Span<felt252>, offset: u8, domain: u32
+        ) -> Span<felt252> {
+            let base = self.compute_base_address(fn_name, params);
+            let mut data = ArrayTrait::new();
+            let mut offset = 0;
+            loop {
+                let value = self._get(domain, base, offset);
+                if value == 0 {
+                    break ();
+                }
+                data.append(value);
+                offset += 1;
+            };
+            data.span()
+        }
+
+        fn _get(
+            self: @ContractState, domain: u32, base: starknet::StorageBaseAddress, offset: u8,
```

Comment:
To support usize instead of u8 we could probably replace base a felt252 and offset by a usize but call the syscall with base = casted(base + offset) and offset = 0